ADALiOS/ADALiOS/ADWebAuthController.h

```diff
@@ -0,0 +1,32 @@
+// Copyright © Microsoft Open Technologies, Inc.
+//
+// All Rights Reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+// ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+// PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache License, Version 2.0 for the specific language
+// governing permissions and limitations under the License.
+
+@class ADAuthenticationError;
+
+#import "ADAuthenticationContext.h"
+
+@interface ADWebAuthController : NSObject
+
++ (ADWebAuthController *)sharedInstance;
+
+//Cancel the web authentication session which might be happening right now
+//Note that it only works if there's an active web authentication session going on
+- (void)cancelCurrentWebAuthSession;
+- (BOOL)cancelCurrentWebAuthSessionWithError:(ADAuthenticationError*)error;
```

Comment:
Do we need to make these two functions class functions rather than instance functions?